param_passing/tutorial.md

```diff
@@ -0,0 +1,561 @@
+  <transparency action="add">0.5</transparency>
+  <pose relative_to="something" action="add">0 0 1 0 0 0</pose>
+</visual>
+```
+
+**Example 3**
+
+To add a new element as a direct child of the included/original model, use `element_id=""`:
+
+```xml
+<!-- //experimental:params -->
+<frame element_id="" name="some_frame" action="add">
+  <pose>0 1 0 0 0 0</pose>
+</frame>
+```
+
+This adds the `//frame[@name="some_frame"]` to the included `//model`
+(i.e., the output would be `//model/frame[@name="some_frame"]`).
+
+---
+
+### Modify action
+
+The `modify` action provides the ability to modify values and/or attributes of elements.
+This action only updates existing elements and does not add or remove them.
+It can update or add attributes to existing elements but can not remove them.
+
+To `modify` elements and/or attributes, the full element path to that element is needed with the new value(s).
+For example, if the original model has a `//visual[@name="visual"]/geometry/sphere[radius=0.5]`
+and modifying the `//radius` is desired, then under `//experimental:params`:
+
+```xml
+<visual element_id="path::to::visual" action="modify">
+  <geometry>
+    <sphere>
+      <radius custom:attr="foo">1.0</radius>
+    </sphere>
+  </geometry>
+</visual>
+```
+
+This modifies the original `//radius` from 0.5 to be 1.0 and adds a new attribute to the element.
+If the original `//visual` has other children elements,
+these elements are left unchanged since they are not provided under `//experimental:params`.
+
+To modify an attribute, the attribute must be specified in the desired element
+and if the element has a value (not children elements) then the value must be provided as well
+even if it is desired to leave the value unchanged (see `modify` example 2).
+If the element does have child elements then only providing the attribute is needed (see example 1).
+If no attributes are provided, the original attributes will be left unchanged.
+Although, it is not possible to remove an attribute, a work around is to assign
+the attribute an empty value (e.g., `//pose[@relative_to=""]`).
+The `@name` attribute can only be modified when the `modify` action is defined in the `@element_id` element.
+
+#### Modify examples
+
+**Example 1**
+
+To `modify` a name attribute only, in this case the original model has `//visual[@name="visual"]`:
+
+```xml
+<!-- //experimental:params -->
+<visual element_id="path::to::visual" name="new_name" action="modify"/>
+```
+
+this only modifies the `@name` of the `//visual` to be `new_name`.
+There are no other listed attributes or child elements so nothing else is changed.
+
+To `modify` child elements and not the name attribute provide
+the full path to the element to be modified with the updated value:
+
+```xml
+<!-- //experimental:params -->
+<visual element_id="path::to::visual" action="modify">
+  <geometry>
+    <sphere>
+      <radius>1.0</radius>
+    </sphere>
+  </geometry>
+</visual>
```

Comment:
the example is basically the same as the one above right (just without the new attribute)?